### PR TITLE
fix: serve PMTiles from R2 for range request support

### PR DESCRIPTION
**Pull request by ashmortar**
## Summary
- Cloudflare Pages static serving doesn't support HTTP Range requests, breaking PMTiles vector tile loading on production
- Added R2 bucket (`dd-tiles`) with a thin API proxy route (`/api/tiles/[...path]`) that properly handles `Range`/`HEAD` requests and returns `206 Partial Content`
- Updated DistrictMap to fetch tiles through the new API route instead of the static path
- Removed misleading `Accept-Ranges` header from `public/_headers`

## Test plan
- [ ] `pnpm lint && pnpm typecheck && pnpm test` all pass
- [ ] Verify on preview deploy: district boundaries render on `/map`
- [ ] Verify range requests work: `curl -sI -H "Range: bytes=0-16383" <preview-url>/api/tiles/districts.pmtiles` returns `206` with `Content-Length` and `Content-Range`
- [ ] After merge, verify at `https://democracy-direct.com/map`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/pages/api/tiles/[...path].ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    if (!object) return notFound();
+
+    return new Response(object.body, {
+      status: 206,
+      headers: {
+        "Content-Length": String(range.length),
+        "Content-Range": `bytes ${range.offset}-${range.offset + range.length - 1}/${head.size}`,
+        "Accept-Ranges": "bytes",
+        "Content-Type": "application/octet-stream",
+        "Cache-Control": CACHE_HEADER,
```
[the remark this answers is not in the document]
Comment (ashmortar) [reply src/pages/api/tiles/[...path].ts]: Added `Vary: Range` to all cacheable responses (HEAD, GET, and 206).

File: src/pages/api/tiles/[...path].ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  return { offset: start, length: end - start + 1 };
+}
+
+const CACHE_HEADER = "public, max-age=604800, s-maxage=2592000, stale-while-revalidate=604800";
+
+const handler: APIRoute = async ({ params, request, locals }) => {
+  const bucket = locals.runtime?.env?.TILES_BUCKET;
+  if (!bucket) {
+    return new Response("Tile storage not configured", { status: 503 });
+  }
```
[the remark this answers is not in the document]
Comment (ashmortar) [reply src/pages/api/tiles/[...path].ts]: Added 15 unit tests in `tests/api/tiles/path.test.ts` covering: parseRange edge cases, 503 (no bucket), 404 (missing path/non-.pmtiles/missing object), 200 full GET, 206 range GET with Content-Range/Content-Length, 416 unsatisfiable range, HEAD with size/Accept-Ranges.
